### ISSUE TITLE
doubled slash in paths in error log

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#8301]
```
2019.12.19 17:33:18.401719 [ 1 ] {} <Error> Application: DB::Exception: Cannot parse definition from metadata file /var/lib/clickhouse/metadata/default//aaaa.sql, error: DB::Exception: Syntax error (in file /var/lib/clickhouse/metadata/default//aaaa.sql): failed at position 21 (line 3, col 1): )
```

etc

**Comment by alexey-milovidov** [comment]
Not an issue.

**Comment by filimonov** [comment]
Rather it's an issue.

Motivation: when people look into logs and see SOME error message related to the filesystem operations, and a message itself has 'malformed' path they start to think that malformed path is the reason of the issue (while it's just a cosmetic).

**Comment by alexey-milovidov** [comment]
Let's fix that.

**Comment by alexey-milovidov** [comment]
You can either check and fix this single case or rewrite all path handling from std::string to std::filesystem::path.

**Comment by mayankkt9** [comment]
Can I work on this ?

**Comment by alexey-milovidov** [comment]
@mayankkt9 Sure. This is yours!

**Comment by alexey-milovidov** [comment]
https://clickhouse.yandex/docs/en/development/developer_instruction/

**Comment by alexey-milovidov** [comment]
@mayankkt9 You can still work on this :smile:

**Comment by alexey-milovidov** [comment]
@mayankkt9 This task is waiting for you!

**Comment by filimonov** [comment]
I think there are more places like this, but that exact issue is fixed in #11900
